libs/entitlement/asm-provider/src/main/java/org/elasticsearch/entitlement/instrumentation/impl/InstrumentationServiceImpl.java

```diff
@@ -40,8 +40,8 @@ public class InstrumentationServiceImpl implements InstrumentationService {
     private static final String OBJECT_INTERNAL_NAME = Type.getInternalName(Object.class);
 
     @Override
-    public Instrumenter newInstrumenter(Class<?> clazz, Map<MethodKey, CheckMethod> methods) {
-        return InstrumenterImpl.create(clazz, methods);
+    public Instrumenter newInstrumenter(Class<?> clazz, Map<MethodKey, CheckMethod> methods, boolean verifyBytecode) {
+        return InstrumenterImpl.create(clazz, methods, verifyBytecode);
```

Comment:
This `verifyBytecode` argument always comes from the system property. Even tests don't pass a different value in here.

I wonder if this factory method itself ought to check the system property?